pkg/models/mto_category.go

```diff
@@ -22,6 +22,12 @@ type MTOCategory struct {
 	ParentID *uuid.UUID `json:"parent_id" db:"parent_id"`
 }
 
+type MTOCategories struct {
+	// TODO (mto) should these fields be pointers? Technically they should never be null
```

Comment:
I could see an argument for both, but if we _do_ make them pointers, it _will_ return a GQL error if we try and return `nil` (since the Schema won't allow it).

That could be viewed as a pro (error if you try and do something you shouldn't) or a con (maybe it'd be better to just accidentally return a zero-valued category?)

Though, in just typing that out, maybe a pointer is better, that way if we _do_ somehow accidentally return a `nil` for one of these fields, we'll get a clear error rather than just returning a zero'd out `MTOCategory` or `MTOSubcategory`, which isn't really helpful.

Reply:
Good call, I'll switch to it. I also like that in favor of passing by reference instead of by value!